app/controllers/CyaMaintainProfileController.scala

```diff
@@ -0,0 +1,111 @@
+ */
+
+package controllers
+
+import cats.data.NonEmptyChain
+import connectors.TraderProfileConnector
+import controllers.actions._
+import models.{CheckMode, NormalMode, TraderProfile, UnexpectedPage}
+import navigation.Navigator
+import pages.{CyaMaintainProfilePage, NiphlNumberUpdatePage, NirmsNumberUpdatePage}
+import play.api.i18n.MessagesApi
+import play.api.mvc.{Action, AnyContent, Call, MessagesControllerComponents}
+import services.AuditService
+import viewmodels.checkAnswers.{HasNirmsSummary, NirmsNumberSummary}
+import viewmodels.govuk.summarylist._
+import views.html.CyaMaintainProfileView
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class CyaMaintainProfileController @Inject() (
+  override val messagesApi: MessagesApi,
+  identify: IdentifierAction,
+  getData: DataRetrievalAction,
+  requireData: DataRequiredAction,
+  val controllerComponents: MessagesControllerComponents,
+  view: CyaMaintainProfileView,
+  traderProfileConnector: TraderProfileConnector,
+  navigator: Navigator,
+  auditService: AuditService
+)(implicit ec: ExecutionContext)
+    extends BaseController {
+
+  private val errorMessage: String = "Unable to update Trader profile."
+  private val continueUrl: Call    = routes.ProfileController.onPageLoad()
+
+  def onPageLoadNirmsNumber(): Action[AnyContent] = (identify andThen getData andThen requireData).async {
+    implicit request =>
+      traderProfileConnector.getTraderProfile(request.eori).flatMap { _ =>
+        request.userAnswers.get(NirmsNumberUpdatePage) match {
+          case Some(nirmsNumberAnswer) =>
+            val list = SummaryListViewModel(
+              rows = Seq(
+                Some(HasNirmsSummary.row(value = true, CheckMode)),
+                NirmsNumberSummary.row(Some(nirmsNumberAnswer))
+              ).flatten
+            )
+            Future.successful(Ok(view(list, routes.CyaMaintainProfileController.onSubmitNirmsNumber)))
+          case None                    =>
+            Future.successful(
+              logErrorsAndContinue(errorMessage, continueUrl, NonEmptyChain.one(UnexpectedPage(NiphlNumberUpdatePage)))
+            )
+        }
+      }
+  }
+
+  def onSubmitNirmsNumber(): Action[AnyContent] = (identify andThen getData andThen requireData).async {
+    implicit request =>
+      traderProfileConnector.getTraderProfile(request.eori).flatMap { traderProfile =>
+        val updatedProfile = traderProfile.copy(nirmsNumber = request.userAnswers.get(NirmsNumberUpdatePage))
+        auditService.auditMaintainProfile(traderProfile, updatedProfile, request.affinityGroup)
+        for {
+          _ <- traderProfileConnector.submitTraderProfile(updatedProfile, request.eori)
+        } yield Redirect(navigator.nextPage(CyaMaintainProfilePage, NormalMode, request.userAnswers))
+      }
+  }
+
+  def onPageLoadNirms(): Action[AnyContent] = (identify andThen getData andThen requireData) { implicit request =>
```

Comment:
This method and those below remain unchanged, I put the new methods at the top. I can reverse this the diff is confusing.